File: app/assets/javascripts/anti-flood.js
define(function () {
  var forms = document.querySelectorAll('form');
  for (var i = 0, l = forms.length ; i < l ; i++) {
    var form = forms.item(i);
    var buttons = form.querySelectorAll('input[type=submit], button:not([type=button])');
    form.addEventListener('submit', (function (buttons) {
      return function () {
        for (var j = 0, m = buttons.length ; j < m ; j++) {
          buttons.item(j).disabled = true;
        }
      }
    })(buttons));
  }
});

File: js-scala/src/main/scala/javascripts/AntiFlood.scala
package javascripts

import scala.js.language.dom.DomReact
import scala.js.language.JS

trait AntiFlood extends JS with DomReact {

  def setupAntiFlood() = for (form <- document.findAll[Element]("form")) {
    val buttons = form.findAll[Input]("input[type=submit], button:not([type=button])")
    for {
      submit <- events.of(Submit, form)
      button <- buttons
    } button.disabled_=(true)
  }

}
